contracts/rootchain/RootChain.sol

```diff
@@ -3,220 +3,315 @@ pragma solidity ^0.4.24;
-            amount: txList[11 + 2 * txPos[2]].toUint(),
-            utxoPos: txPos,
-            created_at: block.timestamp
-        });
+        bytes32 root = childChain[newTxPos[0]].root;
+        bytes32 merkleHash = keccak256(abi.encodePacked(keccak256(txBytes), sigs));
+        bytes32 confirmationHash = keccak256(abi.encodePacked(merkleHash, root));
+        require(exit_.owner == confirmationHash.recover(confirmSignature), "mismatch in exit owner and confirm signature");
+        require(merkleHash.checkMembership(newTxPos[1], root, proof), "incorrect merkle proof");
+
+        // exit successfully challenged
+        balances[msg.sender] = balances[msg.sender].add(minExitBond);
+        totalWithdrawBalance = totalWithdrawBalance.add(minExitBond);
+        
+        depositExits[nonce].state = 2;
+        emit ChallengedDepositExit(nonce, exit_.owner, exit_.amount);
     }
 
-    /// @param txPos [0] Plasma block number in which the challenger's transaction occured
-    /// @param txPos [1] Transaction Index within the block
-    /// @param txPos [2] Output Index within the transaction (either 0 or 1)
-    /// @param newTxPos  Same as the above but the pos of the uxto created by the spend tx
-    function challengeExit(uint256[3] txPos, uint256[3] newTxPos, bytes txBytes, bytes proof, bytes sigs, bytes confirmationSig)
+    // @param txPos            position of the invalid exiting transaction [blkNum, txIndex, outputIndex]
+    // @param newTxPos         position of the challenging transaction [blkNum, txIndex, outputIndex]
+    // @param txBytes          raw transaction bytes of the challenging transaction
+    // @param sigs             signatures of the inputs for this transaction
+    // @param proof            proof of inclusion for this merkle hash
+    // @param confirmSignature signature used to invalidate the invalid exit. Signature is over (merkleHash, block header)
+    function challengeTransactionExit(uint256[3] exitingTxPos, uint256[3] challengingTxPos, bytes txBytes, bytes sigs, bytes proof, bytes confirmSignature)
         public
     {
-        // txBytes verification
         RLPReader.RLPItem[] memory txList = txBytes.toRlpItem().toList();
-        require(txList.length == 15, "incorrect tx list");
-
-        // start-exit verification
-        uint256 priority = 1000000000*txPos[0] + 10000*txPos[1] + txPos[2];
-        uint256[3] memory utxoPos = exits[priority].utxoPos;
-        require(utxoPos[0] == txList[0 + 5 * newTxPos[2]].toUint(), "incorrect blocknum");
-        require(utxoPos[1] == txList[1 + 5 * newTxPos[2]].toUint(), "incorrect tx index");
-        require(utxoPos[2] == txList[2 + 5 * newTxPos[2]].toUint(), "incorrect output index");
+        require(txList.length == 17, "incorrect tx list");
 
-        /*
-           Confirmation sig:
-              txHash, sigs, block header
-          */
+        // invalid transcation should have a pending exit
```

Comment:
This is a little unclear. Perhaps "transaction to be challenged should have a pending exit"

Reply:
done.